src/sentry/api/invite_helper.py

```diff
@@ -211,3 +209,33 @@ def accept_invite(self, user=None):
 
         self.handle_success()
         metrics.incr("organization.invite-accepted", sample_rate=1.0)
+
+    def _needs_2fa(self) -> bool:
+        org_requires_2fa = self.om.organization.flags.require_2fa.is_set
+        user_has_2fa = Authenticator.objects.user_has_2fa(self.request.user.id)
+        return org_requires_2fa and not user_has_2fa
+
+    def _needs_email_verification(self) -> bool:
+        organization = self.om.organization
+        if not (
+            features.has("organizations:required-email-verification", organization)
+            and organization.flags.require_email_verification
+        ):
+            return False
+
+        user = self.request.user
+        primary_email_is_verified = (
```

Comment:
could you use `User.get_verified_emails` here instead? and do we need to check the email returned is the primary? `UserEmail.is_primary`?

Reply:
The filter clause `email=user.email` should have the effect of finding only the primary email (it uses the same logic as in `UserEmail.is_primary`).

Using `User.get_verified_emails` is an improvement. To keep it in a single query, I think we would want to do
```python
user.get_verified_emails().filter(email=user.email).exists()
```
rather than use `UserEmail.is_primary` (which is an application-level check).

Alternatively, we could prioritize clarity over efficiency with
```python
any(email.is_verified() for email in user.get_verified_emails())
```
but I hope that the `primary_email_is_verified` name makes it sufficiently clear.